### ISSUE TITLE
Merge conflicts with bay12 master, take 2

### ISSUE THREAD

**Issue by lmas** [Apollo-Community/ApolloStation#87]
Giving this monster a second try, see #80 for my first attempt.
- [x] .gitignore
- [x] baystation12.dme
- [x] code/controllers/master_controller.dm
- [x] code/controllers/verbs.dm
- [x] code/defines/obj/weapon.dm
- [x] code/game/area/Space Station 13 areas.dm
- [x] code/game/gamemodes/events/biomass.dm
- [x] code/game/gamemodes/events/spacevines.dm
- [x] code/game/gamemodes/game_mode.dm
- [x] code/game/jobs/jobs.dm
- [x] code/game/machinery/autolathe.dm
- [x] code/game/machinery/camera/camera.dm
- [x] code/game/machinery/camera/tracking.dm
- [x] code/game/machinery/doors/blast_door.dm
- [x] code/game/machinery/jukebox.dm
- [x] code/game/machinery/kitchen/processor.dm
- [x] code/game/machinery/kitchen/smartfridge.dm
- [x] code/game/machinery/machinery.dm
- [x] code/game/machinery/suit_storage_unit.dm
- [x] code/game/machinery/vending.dm
- [x] code/game/objects/effects/biomass_rift.dm
- [x] code/game/objects/effects/glowshroom.dm
- [x] code/game/objects/items/weapons/handcuffs.dm
- [x] code/game/objects/items/weapons/manuals.dm
- [x] code/game/objects/items/weapons/storage/uplink_kits.dm
- [x] code/game/objects/items/weapons/tools.dm
- [x] code/game/objects/items/weapons/twohanded.dm
- [x] code/game/objects/random/random.dm
- [x] code/game/objects/structures/girders.dm
- [x] code/game/turfs/simulated/walls.dm
- [x] code/global.dm
- [x] code/modules/admin/topic.dm
- [x] code/modules/clothing/clothing.dm
- [x] code/modules/clothing/head/misc_special.dm
- [x] code/modules/clothing/suits/miscellaneous.dm
- [x] code/modules/clothing/under/miscellaneous.dm
- [x] code/modules/customitems/item_defines.dm
- [x] code/modules/hydroponics/grown_inedible.dm
- [x] code/modules/hydroponics/trays/tray.dm
- [x] code/modules/mining/mine_items.dm
- [x] code/modules/mob/living/carbon/carbon.dm
- [x] code/modules/mob/living/carbon/give.dm
- [x] code/modules/mob/living/carbon/human/examine.dm
- [x] code/modules/mob/living/carbon/human/human.dm
- [x] code/modules/mob/living/living.dm
- [x] code/modules/mob/mob.dm
- [x] code/modules/mob/mob_defines.dm
- [x] code/modules/mob/mob_grab.dm
- [x] code/modules/mob/new_player/new_player.dm
- [x] code/modules/nano/nanomanager.dm
- [x] code/modules/nano/nanoui.dm
- [x] code/modules/paperwork/pen.dm
- [x] code/modules/power/port_gen.dm
- [x] code/modules/projectiles/ammunition/boxes.dm
- [x] code/modules/projectiles/ammunition/bullets.dm
- [x] code/modules/projectiles/gun.dm
- [x] code/modules/projectiles/guns/energy/laser.dm
- [x] code/modules/projectiles/guns/energy/stun.dm
- [x] code/modules/projectiles/guns/launcher/crossbow.dm
- [x] code/modules/projectiles/guns/projectile.dm
- [x] code/modules/projectiles/guns/projectile/automatic.dm
- [x] code/modules/projectiles/guns/projectile/pistol.dm
- [x] code/modules/projectiles/guns/projectile/revolver.dm
- [x] code/modules/projectiles/guns/projectile/rocket.dm
- [x] code/modules/projectiles/guns/projectile/shotgun.dm
- [x] code/modules/projectiles/projectile.dm
- [x] code/modules/projectiles/projectile/bullets.dm
- [x] code/modules/reagents/Chemistry-Machinery.dm
- [x] code/modules/reagents/dartgun.dm
- [x] code/modules/reagents/grenade_launcher.dm
- [x] code/modules/reagents/reagent_containers/food/snacks.dm
- [x] code/modules/reagents/reagent_containers/syringes.dm
- [x] code/modules/reagents/reagent_dispenser.dm
- [x] code/modules/reagents/syringe_gun.dm
- [x] code/modules/virus2/items_devices.dm
- [x] code/setup.dm

All icons/sprites/whatevs that WE added in needs to be moved to the apollo icon dir, so we can merge in these changes (@Kwask, @stuicey you guys knows best what icons changed, think you can fix them?) :
- [x] icons/mob/AI.dmi
- [x] icons/mob/back.dmi
- [x] icons/mob/head.dmi
- [x] icons/mob/items_lefthand.dmi
- [x] icons/mob/items_righthand.dmi
- [x] icons/mob/mask.dmi
- [x] icons/mob/uniform.dmi
- [x] icons/obj/ammo.dmi
- [x] icons/obj/clothing/hats.dmi
- [x] icons/obj/clothing/uniforms.dmi
- [x] icons/obj/custom_items.dmi
- [x] icons/obj/food.dmi
- [x] icons/obj/gun.dmi
- [x] icons/obj/harvest.dmi
- [x] icons/obj/items.dmi
- [x] icons/obj/projectiles.dmi
- [x] icons/obj/syringe.dmi
- [x] icons/obj/weapons.dmi

I assume we can overwrite the maps with these new changes:
- [x] maps/apollo-2.dmm
- [x] maps/apollo-3.dmm
- [x] maps/apollo-4.dmm
- [x] maps/backup/apollo-1.dmm
- [x] maps/exodus-5.dmm

**Comment by lmas** [comment]
See 0d353827a890dba8ae85771e908955fed98d5bee.
- **Code:** Most of the conflicts was fixed this time. Had to comment out some places with `//BOGUS` so I could compile the code, these will have to be fixed properly and/or removed!
- **Icons:** Moved all files that had merge conflicts to the apollo dir and replaced with new bay icons. We will have to strip duplicate icons and update all path references to our custom icons.
- **Maps:** Lot's of undefined object path errors. I simply made backups of the maps and deleted the path definitions. Have to check and fix the maps.
- **Merge status:** the server can be compiled now. Server startup is really slow though and there's some logged errors about calling the `create_random_seed proc`. When connecting a client server seems to hang indefinetily. Have to debug the server.

**Comment by lmas** [comment]
- **Code:** I've fixed a lot more broken code and I can now run the server successfully and play the game. ~~Still some minor code problems I have to fix (feeding carrots to bunnies, runtime errors with asteroid map)~~. Fixed.
- **Maps:** ~~Found more problems with the maps, all engineering turfs are gone and needs to be added in again. Probably the cause for the robotic cryopods hanging the server on initialization.~~ Fixed engineering turfs and runtime warnings with the asteroid map. Only have to add in the missing objects now (try to open the maps in the backup dir and you'll get a list of the missing objects).
- **Icons:** I've managed to fix most of the custom icons now too, ~~except the uniforms. They don't show up when you wear them D:~~ Fixed.
- **Merge status:** We should be good to go now and merge this in if we want. Only have to fix the uniform icons and add in all couches to the maps I think.

**Comment by lmas** [comment]
- **Icons:** I've managed to fix most of the custom icons now too, except the uniforms. They show up when you wear them D:

- **Merge status:** We should be good to go now and merge this in if we want. Only have to fix the uniform icons and add in all couches to the maps I think.

**Comment by Kwask** [comment]
Might want to check if the uniform icons have their _s on the end of them. If they do, try taking it off, if they don't, try putting it on. Bay might've changed how that system works

**Comment by stuicey** [comment]
The _s at the end is to signal that it is the south directional icon. It is only used for /mob icons.

**Comment by Kwask** [comment]
Oh, I always thought it signified "standing", since on some other codebases they have seperate icons for standing and laying, and the standing ones have _s at the end

Either way, awesun is talking about the /mob icons, since they're not displaying when they're being worn.

**Comment by lmas** [comment]
For the hell of it I tried adding an _s to one suit, didn't work.

**Comment by Kwask** [comment]
Make sure in the code that the item_state doesn't have a _s at the end of it

**Comment by lmas** [comment]
```
/obj/item/clothing/under/apollo/camo
    icon = 'icons/apollo/clothing/uniforms.dmi'
    name = "camo jumpsuit"
    desc = "Doesn't exactly blend with space, now does it?"
    icon_state = "camo"
    item_state = "camo"
    item_color = "camo"

/icons/apollo/clothing/uniforms.dmi
    camo icon name = "camo"
```

I got nuthing

**Comment by Kwask** [comment]
What about the icons/mob/uniform.dmi? That's where item_state comes from.

How it used to work was that the icon in mob/uniform.dmi has _s at the end of it, and the item_state would have the icon name without the _s

**Comment by lmas** [comment]
Shit, there's mob/uniform.dmi (where all icon names ends in _s or _d_s) and then there's obj/uniforms.dmi (notice it's uniform**s**.dmi and there's no _s at the end of the icon names).

I accidently deleted the mob/uniform.dmi. :bowtie:

**Comment by Kwask** [comment]
whoopsies :baby: 

Thankfully that one file is easy to revert or steal from the main branch

**Comment by lmas** [comment]
Found some icon override, uniforms works. We can merge in this branch now without too much trouble. The maps can be fixed over time, as long as we keep our customisations.

Some small [runtime errors](https://github.com/Baystation12/Baystation12/issues/8536), gonna have to sync up with bay over the next days for the fix.

**Comment by lmas** [comment]
crap, merged in master and now getting:

	runtime error: Cannot read null.grid_sensors
	proc name: check warnings (/obj/machinery/computer/power_monitor/proc/check_warnings)
	  source file: sensor_monitoring.dm,78
	  usr: null
	  src: the telecoms power monitoring (/obj/machinery/computer/power_monitor)
	  call stack:
	the telecoms power monitoring (/obj/machinery/computer/power_monitor): check warnings()
	the telecoms power monitoring (/obj/machinery/computer/power_monitor): process()
	/datum/controller/game_control... (/datum/controller/game_controller): process machines process(1, 6603)
	/datum/controller/game_control... (/datum/controller/game_controller): process()

**Comment by stuicey** [comment]
Looks like you've reverted master controller to original bay

**Comment by lmas** [comment]
Oops.. kwask broke it :bowtie:

**Comment by Kwask** [comment]
eyy, a fuka u

**Comment by lmas** [comment]
With 2803e00571b9542be60b631606801edd65adcfd8 I reverted the master controller. I'm not too sure about removing the new subsystems bay added in. Sure, it might speed up things but on the other hand maybe it's required? I haven't looked too much into it yet and haven't run the server for any longer periods so I have no clue.

**Comment by lmas** [comment]
Are we ready for the merge or what? I think this potato is hot enough now.

**Comment by Kwask** [comment]
I think it's positively radioactive :warning: :sweet_potato: 

here goes nothing (or more accurately, everything we hold near and dear)#93

**Comment by JMMJ01** [comment]
🔜

**Comment by Kwask** [comment]
fixed in 72a2040cc6b25dcde032b98edd616413ec864fa8
